src/codeflare_sdk/cluster/cluster.py

```diff
@@ -644,7 +645,40 @@ def get_cluster(cluster_name: str, namespace: str = "default"):
     for rc in rcs["items"]:
         if rc["metadata"]["name"] == cluster_name:
             mcad = _check_aw_exists(cluster_name, namespace)
-            return Cluster.from_k8_cluster_object(rc, mcad=mcad)
+            # Case where mcad is true we can gather the domain name from the ingress specified in the appwrapper.
+            if mcad:
+                try:
+                    config_check()
+                    api_instance = client.CustomObjectsApi(api_config_handler())
+                    appwrapper = api_instance.get_namespaced_custom_object(
+                        group="workload.codeflare.dev",
+                        version="v1beta1",
+                        namespace=namespace,
+                        plural="appwrappers",
+                        name=cluster_name,
+                    )
+                    ingress_host = appwrapper["spec"]["resources"]["GenericItems"][1][
+                        "generictemplate"
+                    ]["spec"]["rules"][0]["host"]
+                except Exception as e:  # pragma: no cover
+                    return _kube_api_error_handling(e, print_error=False)
+            else:
+                # Case where mcad is false we get the host from the created ingress.
+                try:
+                    config_check()
+                    api_instance = client.NetworkingV1Api(api_config_handler())
+                    ingresses = api_instance.list_namespaced_ingress(namespace)
+                    for ingress in ingresses.items:
+                        if ingress.metadata.labels["ingress-owner"] == cluster_name:
```

Comment:
Have you consider using [Owner functionality](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/) instead of custom label?